R/drug_parameters.R

```diff
@@ -3,37 +3,104 @@
+#' @details Default parameters, from L to R, are: drug_efficacy: 1, drug_rel_c: 0.5, drug_prophylaxis_shape: 11.3, drug_prophylaxis_scale: 10.6
+#' @export
+white_ACT_params_vivax <- list("vivax", c(0.899, 0.5, 11.3, 10.6))
+
+#' @title Preset parameters for the ACT-PQ drug
+#' @description Efficacy from SI of White et al., DOI: 10.1371/journal.pmed.1003535 (2021)
+#' BS shape and scale consistent with AL prophylaxis in falciparum
+#' LS shape are scale consistent with a 14 day PQ treatment, with rapid decay.
+#' @details Use a vector of preset parameters for the ACT-PQ drug (Artemisinin combination therapy and primaquine)
+#' @details Default parameters, from L to R, are: drug_efficacy: 1, drug_rel_c: 0.5, drug_prophylaxis_shape: 11.3, drug_prophylaxis_scale: 10.6,
+#' drug_hypnozoite_efficacy: 0.7, drug_hypnozoite_prophylaxis_shape: 11.3, drug_hypnozoite_prophylaxis_scale: 10.6
+#' @export
+white_ACT_PQ_params_vivax <- list("vivax", c(1, 0.5, 11.3, 10.6, 0.7, 11.3, 10.6))
+
+#' @title Preset parameters for the ChlQ drug
+#' @description Efficacy from SI of White et al., DOI: 10.1371/journal.pmed.1003535 (2021)
+#' BS shape and scale consistent with AL prophylaxis in falciparum
+#' LS shape are scale consistent with a 14 day PQ treatment, with rapid decay.
+#' @details Use a vector of preset parameters for the ChlQ drug (chloroquine)
+#' @details Default parameters, from L to R, are: drug_efficacy: 1, drug_rel_c: 0.5, drug_prophylaxis_shape: 11.3, drug_prophylaxis_scale: 10.6,
+#' drug_hypnozoite_efficacy: 1, drug_hypnozoite_prophylaxis_shape: 11.3, drug_hypnozoite_prophylaxis_scale: 10.6
+#' @export
+white_ACT_TQ_params_vivax <- list("vivax", c(1, 0.5, 11.3, 10.6, 1, 5, 30))
+
 
 #' @title Parameterise drugs to use in the model
 #'
 #' @param parameters the model parameters
 #' @param drugs a list of drug parameters, can be set using presets
 #' @export
 set_drugs <- function(parameters, drugs) {
-  stopifnot(is.list(drugs))
+  stopifnot("is.list(drugs) is not TRUE" = is.list(drugs))
+  stopifnot("is.list(drugs) is not TRUE" = any(unlist(lapply(drugs, is.list))))
   keys <- c(
     'drug_efficacy',
     'drug_rel_c',
     'drug_prophylaxis_shape',
-    'drug_prophylaxis_scale'
+    'drug_prophylaxis_scale',
+    # hypnozoite parameters
+    'drug_hypnozoite_efficacy',
+    'drug_hypnozoite_prophylaxis_shape',
+    'drug_hypnozoite_prophylaxis_scale'
   )
+  length_drug_parameters <- max(sapply(drugs, function(d){length(d[[2]])}))
   for (drug in seq_along(drugs)) {
-    for (i in seq_along(drugs[[drug]])) {
-      parameters[[keys[[i]]]] <- c(parameters[[keys[[i]]]], drugs[[drug]][[i]])
+    stopifnot("parasite and drug parameters do not align" = parameters$parasite == drugs[[drug]][[1]])
+    for (i in 1:length_drug_parameters) {
+      if(i <= length(drugs[[drug]][[2]])){
+        parameters[[keys[[i]]]] <- c(parameters[[keys[[i]]]], drugs[[drug]][[2]][[i]])
+      } else {
+        parameters[[keys[[i]]]] <- NA
```

Comment:
Oh I see.

Yes it seems like we could rename `falciparum_names` -> `drug_param_names` and `vivax_names` -> `liver_stage_drug_param_names` or something more concise 😄 and relax the validation accordingly.

On the issue of the user putting in the wrong drug parameters in:

If the researcher is using the default parameters, I would expect them to know if the drug works at the liver stage or not? And for some drugs, whether they treat vivax or falciparum.

And if they are constructing a drug parameter list from their own research, it feels like circular validation to ask them what parasite it targets so that we can validate it against the parasite they specified using `set_parasite`. 

Renaming any conflicting default drug parameters to `drug_name_falciparum` and `drug_name_vivax` seems like a fair response to me.

Reply:
I've renamed and switched back to the previous iteration - where drug parameters are a single list (nest push).